### PR TITLE
[Bugfix] Keep speculative mRoPE deltas shaped (B, 1) for multimodal batches

### PR DESCRIPTION
**Pull request by alivepea**
## Motivation

Speculative decoding (NEXTN / EAGLE) crashes on concurrent multimodal audio requests when `mrope_position_delta` is a 1-D tensor (as returned by Qwen3-ASR).

In `ForwardBatch.compute_spec_mrope_positions`, the mixed-batch path did:

```python
mm_inputs[i].mrope_position_delta.squeeze(0)  # (1,) -> 0-d
torch.stack(...).to(device)                   # -> (B,)
```

Draft positions are `(B, num_draft_tokens)`. Broadcasting `(B, num_draft) + (B,)` aligns on the **last** dim, so:

- `B == 1`: works by accident
- `B != num_draft_tokens` (typical concurrent ASR): `RuntimeError` size mismatch
- `B == num_draft_tokens`: may silently mis-broadcast

Text-only path already used the correct `(B, 1)` shape.

## Modifications

- `forward_batch_info.py`: always reshape per-request deltas to `(B, 1)` before adding to draft positions.
- `qwen3_asr.py`: return `mrope_position_delta` as `(1, 1)` (defense in depth; matches other VL processors).

## Accuracy Tests

Not expected to change text-only outputs. Verified on H20 with NEXTN (`--speculative-num-draft-tokens 4`) + concurrent `/v1/chat/completions` audio requests:

| Concurrency | Requests | Result |
|-------------|----------|--------|
| 2 | 16/16 | pass (previously crash) |
| 3 | 15/15 | pass (previously crash) |
| 4 | 16/16 | pass |
| 8 | 24/24 | pass |

Server stayed healthy; no `compute_spec_mrope_positions` / `SIGQUIT` after the fix.

## Speed Tests and Profiling

N/A (correctness fix; no intended perf change).

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

CC speculative-decoding / model-executor reviewers: @hnyls2002 @Qiaolin-Yu @merrymercy @Ying1123

Please help `/tag-and-rerun-ci` if you have CI permission.

Made with [Cursor](https://cursor.com)

---
### CI States

Latest PR Test (Base): :x: [Run #33308938205](https://github.com/sgl-project/sglang/actions/runs/33308938205)
Latest PR Test (Extra): :x: [Run #33308938102](https://github.com/sgl-project/sglang/actions/runs/33308938102)
Latest PR Test (AMD ROCm 7.2): :x: [Run #33308938217](https://github.com/sgl-project/sglang/actions/runs/33308938217)

### CODE REVIEW

**Comment by alivepea** [comment]
CI gate failed because this PR does not yet have the `run-ci` label (`Missing required label 'run-ci'`).

I do not have CI trigger permission. Could a maintainer please run:

```
/tag-and-rerun-ci
```

Thanks!

File: python/sglang/srt/multimodal/processors/qwen3_asr.py [hunk 1]
```diff
@@ -57,7 +57,9 @@ def compute_mrope_positions(self, input_ids, mm_items):
             seq_len = input_ids.shape[-1] if input_ids.dim() > 1 else input_ids.shape[0]
         positions = torch.arange(seq_len, dtype=torch.long)
         mrope_positions = positions.unsqueeze(0).expand(3, -1).clone()
-        return mrope_positions, torch.tensor([0], dtype=torch.long)
+        # Shape (1, 1) so speculative mRoPE path keeps per-request deltas as
+        # (B, 1) after stacking (1-D [0] becomes (B,) and breaks draft broadcast).
+        return mrope_positions, torch.zeros((1, 1), dtype=torch.long)
```
Comment (mickqian) [inline python/sglang/srt/multimodal/processors/qwen3_asr.py]: could you make `torch.zeros((1, 1), dtype=torch.long)` a constant so that it could be reused across all processors?

**Comment by alivepea** [comment]
@mickqian Addressed your comment: extracted `ZERO_MROPE_POSITION_DELTA` in `base_processor.py` and switched Qwen3-ASR to `ZERO_MROPE_POSITION_DELTA.clone()` so other processors can reuse the same constant.

**Comment by alivepea** [comment]
Hi @mickqian — thank you for the earlier review suggestion!

I've addressed your comment in the follow-up commit (`Share ZERO_MROPE_POSITION_DELTA for multimodal processors`): the shared `(1, 1)` constant now lives in `base_processor.py`, and Qwen3-ASR uses `ZERO_MROPE_POSITION_DELTA.clone()`.

Whenever you have a moment, could you please take another look and Approve if it looks good?

Also kindly requesting CI help from maintainers — this PR is still missing the `run-ci` label, so the gate fails before real tests run:

```
/tag-and-rerun-ci
```

CC model-executor / speculative oncalls: @hnyls2002 @merrymercy @Ying1123 @Fridge003 @ispobock

Thanks so much for your time!

**Review by mickqian — APPROVED** [review]
